buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -91,12 +91,17 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
     }
 
 
-fun splitIntoBuckets(sourceFiles: List<File>, numberOfSplits: Int): List<List<File>> =
-    if (sourceFiles.size % numberOfSplits == 0) {
-        sourceFiles.chunked(sourceFiles.size / numberOfSplits)
-    } else {
-        sourceFiles.chunked(sourceFiles.size / numberOfSplits + 1)
+/**
+ * Distributed test requires all dependencies to be declared
+ */
+fun Project.integrationTestUsesSampleDir(vararg sampleDirs: String) {
+    tasks.withType<IntegrationTest>() {
+        systemProperty("autoTestedSamplesDeclared", "true")
```

Comment:
Should we be more specific which samples are declared?

Reply:
Sure, yes.